vision/lidar/view_recording.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/python3
+"""View data from a pcap file recorded with `record_data.py`."""
+
+import os
+import sys
+import argparse
+
+import cv2
+import numpy as np
+from nptyping import NDArray, Shape, UInt8, UInt32
+
+from ouster import client, pcap
+from ouster.sdk.examples.pcap import pcap_3d_one_scan
+
+
+HSCALE_DEFAULT = 2  # horizontal scale of image viewer
+VSCALE_DEFAULT = 8  # vertical scale of image viewer
+
+
+def view_recording(
+    pcap_file: str,
+    meta_file: str,
+    hscale: int = HSCALE_DEFAULT,
+    vscale: int = VSCALE_DEFAULT,
+    **viz_flags: bool,
+) -> None:
+    """View a pcap file recorded with `record_data.py`.
+
+    Parameters
+    ----------
+    pcap_file : str
+        Path to pcap file
+    meta_file : str
+        Path to metadata json file
+    hscale : int, optional, default=HSCALE_DEFAULT
+        Horizontal scale of image viewer, by default HSCALE_DEFAULT
+    vscale : int, optional, default=VSCALE_DEFAULT
+        Vertical scale of image viewer, by default VSCALE_DEFAULT
+    **viz_flags : dict[str, bool]
+        Flags for which visualizations to show.
+        Options are:
+            "range": Range visualization
+            "reflectivity": Reflectivity visualization
+            "signal": Signal visualization
+            "nearir": Near infrared visualization
+            "pointcloud": 3D point cloud visualization of first frame
+    """
+
+    metadata: client.SensorInfo | None = None
+    with open(meta_file, mode="r", encoding="UTF-8") as json:
+        metadata = client.SensorInfo(json.read())
+
+    source: pcap.Pcap = pcap.Pcap(pcap_path=pcap_file, info=metadata)
+
+    if viz_flags.get("pointcloud"):
+        pcap_3d_one_scan(
+            source, metadata, num=0
+        )  # uncomment to view 3d point cloud of first frame of recording
```

Comment:
This isn't commented so idk what this comment is referring to. If this is meant to be uncommented, should remove the comment about uncommenting.